### PR TITLE
Complete Sprint 8 Prep Task 3: Research Option Statement Syntax

### PR DESCRIPTION
**Pull request by jeffreyhorn**
## Summary

Comprehensive research of GAMS option statement syntax to guide Sprint 8 implementation. Validated that basic integer options cover 100% of GAMSLib usage and confirmed 6-8 hour implementation estimate.

## Deliverables

**Created:**
- `OPTION_STATEMENT_RESEARCH.md` (750+ lines)
  - GAMS documentation survey
  - GAMSLib usage analysis (3 models use options)
  - Lark grammar design (5 rules)
  - Test fixture planning (8 cases)
  - Implementation effort validation (7.5 hours)

**Modified:**
- `KNOWN_UNKNOWNS.md` (verified unknowns 1.1, 1.2, 1.3)
- `PREP_PLAN.md` (Task 3 marked COMPLETE)
- `CHANGELOG.md` (added Task 3 entry)

### GAMS Documentation Survey
- 6 value type categories documented
- Basic syntax: `option key = value [, key = value]*;`
- No parser behavior dependencies
- Sprint 8 scope: Integer values + boolean on/off keywords

### GAMSLib Usage Analysis
- **3 of 10 models** (30%) use options: mhw4dx, maxmin, mingamma
- All basic integer types: limrow, limcol, decimals
- Multi-option statements in 2 of 3 models
- **Sprint 8 unlock:** mhw4dx.gms only (+10%: 2/10 → 3/10)

### Grammar Design
```lark
option_stmt: ("option"i | "options"i) option_list SEMI
option_list: option_item ("," option_item)*
option_item: ID "=" option_value | ID
option_value: NUMBER | "on"i | "off"i
```

### Test Fixtures
- 8 test cases planned (5 positive + 3 edge cases)
- Single option, multi-option, multiple statements, options in context
- Edge cases: empty list, missing semicolon, case insensitivity

### Implementation Effort
- **Detailed breakdown:** 7.5 hours (within 6-8 hour estimate)
  - Grammar: 1h, AST: 1h, Fixtures: 3h, Testing: 1.5h, Docs: 1h
- **Risk:** Low (grammar extension only, no semantic processing)

### 1.1: Is semantic handling straightforward?
✅ **VERIFIED:** Mock/store approach is sufficient. GAMS options control output/solver settings, not parsing behavior. No behavioral implementation required for Sprint 8.

### 1.2: What is GAMSLib option scope?
✅ **VERIFIED:** Basic integer options cover 100% of GAMSLib usage. No advanced features (projection, per-identifier) in GAMSLib models.

### 1.3: Do options unlock mhw4dx.gms?
✅ **VERIFIED:** Option statements are sole blocker. Manual inspection confirms no secondary errors. +10% parse rate confirmed (2/10 → 3/10).

## Sprint 8 vs Sprint 8b Scope

**Sprint 8 (Confirmed):**
- Integer options (limrow, limcol, decimals)
- Boolean on/off keywords
- Multi-option statements
- Mock/store semantic handling
- **Unlocks:** mhw4dx.gms (+10%)
- **Effort:** 6-8 hours

**Sprint 8b (Deferred):**
- Per-identifier syntax (`:\ operator)
- Projection/permutation (`<`, `<=`, `>`)
- Float/string values
- Semantic processing
- **Unlocks:** 0 models
- **Effort:** 4-6 hours

## Impact on Sprint 8

- ✅ Implementation ready (grammar designed, fixtures planned)
- ✅ Effort estimate confirmed (7.5 hours within 6-8 hour range)
- ✅ High confidence in +10% unlock
- ✅ Exceeds conservative 25% target (reaches 30% with options alone)
- ✅ No hidden complexity discovered
- ✅ Low risk implementation path

## Acceptance Criteria

- [x] Option syntax patterns documented
- [x] GAMSLib usage analyzed
- [x] Lark grammar designed
- [x] Test fixtures identified (8 cases)
- [x] Implementation effort validated (7.5 hours)
- [x] Sprint 8 vs Sprint 8b scope defined
- [x] Semantic handling approach decided (mock/store)

## Next Steps

Ready for Copilot review and merge.

### CODE REVIEW

File: docs/planning/EPIC_2/SPRINT_8/OPTION_STATEMENT_RESEARCH.md [hunk 1]
```diff
@@ -0,0 +1,769 @@
...
+$onText
+Fixture: Options in Context
+Tests: Option statement between other declarations
+Expected: Parse successfully, options don't break surrounding statements
+$offText
+
+Set i / 1*10 /;
+Scalar x;
+
+option limcol = 0, limrow = 0;
```
[the remark this answers is not in the document]
Comment (jeffreyhorn) [reply docs/planning/EPIC_2/SPRINT_8/OPTION_STATEMENT_RESEARCH.md]: Fixed in 055c5a9. Standardized to lowercase 'limcol' at line 476 in OPTION_STATEMENT_RESEARCH.md.

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,153 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+- **Scope rules:** Sequential execution, reassignment allowed, no parser behavior dependencies
+- **Edge cases:** No expressions allowed (literals only), option names not reserved words
+- **Sprint 8 scope:** Integer values + boolean on/off keywords (covers 100% of GAMSLib)
+
+**GAMSLib Usage Analysis:**
+- **3 of 10 models** (30%) use option statements
+  - mhw4dx.gms (line 37, 47): `option limCol = 0, limRow = 0;` + `option decimals = 8;`
+  - maxmin.gms (line 86): `option limCol = 0, limRow = 0;`
+  - mingamma.gms (line 43): `option decimals = 8;`
+- **All options are basic integer types:** limrow/limCol (0 = suppress), decimals (0-8 display precision)
```
[the remark this answers is not in the document]
Comment (jeffreyhorn) [reply CHANGELOG.md]: Fixed in 055c5a9. Standardized to lowercase 'limrow' in CHANGELOG.md.

File: docs/planning/EPIC_2/SPRINT_8/KNOWN_UNKNOWNS.md [hunk 3]
```diff
@@ -197,8 +216,39 @@ grep -n "^option\|^ option" tests/fixtures/gamslib/*.gms
...
+**Date:** 2025-11-17
+
+**Findings:**
+1. ✅ **3 of 10 GAMSLib models use option statements (30%)**
+   - mhw4dx.gms: `option limCol = 0, limRow = 0;` (line 37), `option decimals = 8;` (line 47)
+   - maxmin.gms: `option limCol = 0, limRow = 0;` (line 86)
+   - mingamma.gms: `option decimals = 8;` (line 43)
+
+2. ✅ **All options are basic integer types:**
+   - limrow/limCol: Integer (0 = suppress listing)
```
[the remark this answers is not in the document]
Comment (jeffreyhorn) [reply docs/planning/EPIC_2/SPRINT_8/KNOWN_UNKNOWNS.md]: Fixed in 055c5a9. Standardized to lowercase 'limcol' and 'limrow' throughout KNOWN_UNKNOWNS.md.
